src/shared/dlt_common.c

```diff
@@ -4149,13 +4149,20 @@ int16_t dlt_getloginfo_conv_ascii_to_uint16_t(char *rp, int *rp_count)
     if ((rp == NULL) || (rp_count == NULL))
         return -1;
 
+#if (BYTE_ORDER == BIG_ENDIAN)
+    num_work[0] = *(rp + *rp_count + 0);
+    num_work[1] = *(rp + *rp_count + 1);
+    num_work[2] = *(rp + *rp_count + 2);
+    num_work[3] = *(rp + *rp_count + 3);
+#else
```

Comment:
The endianess of the payload of the getinfolog message is not about the host endianness (as indicated by the macro BYTE_ORDER). The MSBF bit of the dlt message have to be evaluated here. Unfortunately it is not pased down to this function.

As per autosar spec R20-11 Log and Trace Protocol Specification:
[PRS_Dlt_00604] If the MSBF bit is set to ‘0’, the payload data is encoded in little endian format. ⌋ (RS_LT_00002, RS_LT_00016)
[PRS_Dlt_00605] If the MSBF bit is set to ‘1’, the payload data is encoded in big endian format. ⌋ (RS_LT_00002, RS_LT_00016)

Unfortunately, the information of the MSBF bit is not passed to this function. As the function is part of dlt_common.h and therefore part of the public DLT-API, we cannot change it without major version update of DLT. I would propose to add new functions like:
dlt_getloginfo_conv_ascci_to_*_msbf(char *rp, int *rp_count, int msbf);
to maintain backward API compatiblity.

@minminlittleshrimp , please give me a call before you start implementing.

Reply:
Hello @minminlittleshrimp , hello @LocutusOfBorg ,

i would like to be a bit more pragmatic here. Lets keep the payload as little endian for big and little endian machines.
Then we could convert all 16bit- values in the unit test using htole16() and we are done and unblock the build of the  Debian packages.

Adding now an API-change would delay the solution until we release a DLT 3.0 (which is not even on the roadmap).

I would keep a bug ticket open that big-endian payloads are not supported.

What do you think?

Reply:
Hello @michael-methner 
I think it is good enough for now.
I soon apply the solution for review.
Regards